keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -1042,18 +1044,27 @@ class SqlDeliveryConfigRepository(
         select(DELIVERY_CONFIG.UID, DELIVERY_CONFIG.NAME)
           .from(DELIVERY_CONFIG, DELIVERY_CONFIG_LAST_CHECKED)
           .where(DELIVERY_CONFIG.UID.eq(DELIVERY_CONFIG_LAST_CHECKED.DELIVERY_CONFIG_UID))
+          .and(
+            DELIVERY_CONFIG_LAST_CHECKED.LOCKED_BY.isNull
+              .or(DELIVERY_CONFIG_LAST_CHECKED.LOCKED_AT.lessOrEqual(cutoff))
+          )
```

Comment:
This clause allows the delivery config to be reconsidered for checking if the lock is stale. I figured using the same cutoff made sense rather than overcomplicating things.